bingo.js

```diff
@@ -602,9 +675,119 @@ function setSquareColor(square, colorClass)
 {
 	ALL_COLOURS.forEach(c => square.removeClass(c));
 	square.addClass(colorClass);
+	updateCurrentSheetProgress();
 }
 
-function copySeedToClipboard(id, event)
+function loadBingoProgress(sheetToLoad)
+{
+	/* 
+	 * Expected input for sheetToLoad is [1,3,2,5,..]
+	 * where index of array is numeric value of square id
+	 * and cell value is index of ALL_COLOURS 
+	 */
+	if (typeof sheetToLoad !== 'undefined' && sheetToLoad.length) {
+		forEachSquare((i, square) => {
+			$.each(ALL_COLOURS, function(index, colorClass){
+	       square.removeClass(colorClass);
+	    });
+			square.addClass(ALL_COLOURS[sheetToLoad[i]]);
+		});
+	}
+}
+
+var sheetProgressDebounce = null;
+function updateCurrentSheetProgress()
+{
+	if(sheetProgressDebounce) clearTimeout(sheetProgressDebounce);
+	sheetProgressDebounce = setTimeout(function() {
+			updateCurrentSheetProgressDebounced()
+	}, 100);
+}
+
+function updateCurrentSheetProgressDebounced()
+{
+	if (! isLocalStorageIsAvailable()) {
+		return;
+	}
+
+	processSheetAndStoreAs(SHEET_PROGRESS_KEY + generateNewUrlParam().replace(/-\d-\d/, ''));
+}
+
+function processSheetAndStoreAs(localStorageKey)
+{
+	var squares = generateSquareProgress();
+	result = {
+		squares: squares,
+		timestamp: Date.now(),
+		haveMarkedSquares: squares.reduce(function(a,b){return !!a | !!b})
+	};
+	
+	// save only if current sheet have marked squares or already in storage
+	if (result.haveMarkedSquares || localStorage.getItem(localStorageKey)) {
+		pushNewLocalSetting(localStorageKey, JSON.stringify(result));
+	}
+	// update &progress param is such exists
+	if (gup(URL_PARAM_PROGRESS)) {
+		pushNewUrl(generateNewUrlParamWithProgress(squares));
+	}
+	checkAndRemoveOldSaves();
+}
+
+function checkAndRemoveOldSaves()
+{
+	var saves = getSaves();
+	if (saves.length > SHEET_PROGRESS_SAVE_LIMIT) {
+		var oldestSave = saves.slice(-1).pop();
```

Comment:
Am I missing something? It seems that call to `.slice(-1)` isn't needed here: https://stackoverflow.com/a/19544496/1083697

Reply:
No,.. that was just strange late-night decision of mine 🤔